FurCoNZ/Controllers/OrderController.cs

```diff
@@ -45,16 +46,65 @@ public async Task<IActionResult> Index(CancellationToken cancellationToken)
         }
 
         [HttpPost]
+        [ValidateAntiForgeryToken]
         public async Task<IActionResult> Index(OrderIndexViewModel orderIndexViewModel, CancellationToken cancellationToken)
         {
             if (ModelState.IsValid)
             {
                 var viewModel = new List<TicketDetailViewModel>();
+                var ticketTypes = await _orderService.GetTicketTypesAsync(false, cancellationToken);
+
+                var ticketIndex = 0;
+                foreach (var ticketTypeOrdered in orderIndexViewModel.AvailableTicketTypes.Where(x => x.Value.QuantityOrdered > 0))
+                {
+                    var ticketTypeId = ticketTypeOrdered.Key;
+                    var quantityOrderedForTicketType = ticketTypeOrdered.Value.QuantityOrdered;
+
+                    for (var i = 0; i < quantityOrderedForTicketType; i++)
+                    {
+                        viewModel.Add(new TicketDetailViewModel
+                        {
+                            Id = ++ticketIndex,
+                            TicketTypeId = ticketTypeId,
+                            TicketTypeName = ticketTypes.FirstOrDefault(x => x.Id == ticketTypeId).Name,
```

Comment:
`FirstOrDefault` implies a a ticket type my not be found. A null reference exception may be thrown in a rare case. 
Either check for null or Use `First`. The later will thrown an exception if no entry is found, which may be more informative. Though that is extreamly unlikely unless the database was edited as a user lands on this page.

Reply:
Resolved in 684031c